### DISCUSSION TITLE
Not sure if `from`, `to` in `diff.FilePatch.Files()` is reversed

### DISCUSSION THREAD

**Discussion by zregvart** [go-git/go-git#1061] [Q&A] [answered]
I'm trying to go through the log of a repository and output if a file has been added, deleted or modified.

[This](https://github.com/zregvart/go-git-patch/blob/eee9efe5c3b8311253587860099f5e15331c90fb/main.go#L25-L36) is the code should print a line with `+`, `-` or `~` on the first column of the line if a file has been added, deleted or modified in a particular commit.

When looking at a commit that [added](https://github.com/zregvart/go-git-patch/commit/09e0083ccdea187c3c5a4441884e2f505f5f604c) a file, the `to` is `nil` and `from` has a `diff.File` instance, and reading the [documentation](https://pkg.go.dev/github.com/go-git/go-git/v5@v5.11.0/plumbing/format/diff#FilePatch) I though it should be the other way around:

> If the patch creates a new file, "from" will be nil. If the patch deletes a file, "to" will be nil.

To see this for yourself do this:

```shell
$ git clone https://github.com/zregvart/go-git-patch.git
$ cd go-git-patch
$ go run main.go
```

This prints:

```
eee9efe5c3b8311253587860099f5e15331c90fb
~ README.md
09e0083ccdea187c3c5a4441884e2f505f5f604c
~ README.md
648875269a3450e467c8948d338767ca1f07b481
+ go.mod
+ go.sum
+ main.go
```

And I'm expecting it to print:

```
eee9efe5c3b8311253587860099f5e15331c90fb
~ README.md
09e0083ccdea187c3c5a4441884e2f505f5f604c
+ README.md
648875269a3450e467c8948d338767ca1f07b481
+ go.mod
+ go.sum
+ main.go
```

That is, in `09e0083ccdea187c3c5a4441884e2f505f5f604c` the `README.md` is added and should be prefixed with `+`.

What am I doing wrong?

**Comment by AriehSchneier** [comment, marked as answer]
I think your lines [51 and 52](https://github.com/zregvart/go-git-patch/blob/eee9efe5c3b8311253587860099f5e15331c90fb/main.go#L51) are incorrect/back to front. [Patch](https://pkg.go.dev/github.com/go-git/go-git/v5@v5.11.0/plumbing/object#Commit.Patch) gives you the patch from 1 commit to the next, so you want from the parent to the child, ie your code should look something like:
```
			parent := result(c.Parent(0))
			p := result(parent.Patch(c))
```

**Reply by zregvart** [reply]
That was it, thank you!
